docs/assembly.rst

```diff
@@ -906,8 +940,16 @@ Pseudocode::
 Opcode Stream Generation
 ------------------------
 
-During opcode stream generation, we keep track of the current stack height,
-so that accessing stack variables by name is possible.
+During opcode stream generation, we keep track of the current stack height
+in a counter,
+so that accessing stack variables by name is possible. The stack height is modified with every opcode
+that modifies the stack and with every label that is annotated with a stack
+adjustment. Every time a new
+local variable is introduced, it is registered together with the current
+stack height. If a variable is accessed (either for copying its value or for
```

Comment:
Am I correct that the EVM translation assumes an EVM stack layout
```
| excess items in the innermost block |
| local variables in the innermost block |
...
| excess items in the second outermost block |
| local variables in the second outermost block |
| excess items in the outermost block |
| local variables in the outermost block |
```
?

Am I also correct that `label[]:` has the effect of removing the excess items in the innermost block in the assumed EVM stack in the EVM translation?

Since the language is defined by EVM code-generation, I need a picture of EVM stacks in order to understand anything in this language.